api/services/reporting/report_generator.py

```
@@ -0,0 +1,207 @@
+            template_obj = self.env.get_template(template)
+            html_content = template_obj.render(**context)
+
+            # Generate PDF in thread pool (blocking operation)
+            loop = asyncio.get_event_loop()
+            pdf_bytes = await loop.run_in_executor(
+                None,
+                self._generate_pdf_sync,
+                html_content
+            )
+
+            logger.info(f"PDF generated successfully ({len(pdf_bytes)} bytes)")
```

Ruff (0.14.5)

Logging statement uses f-string

(G004)